cfme/physical/physical_rack.py

```diff
@@ -0,0 +1,336 @@
+# -*- coding: utf-8 -*-
+"""A model of an Infrastructure PhysicalRack in CFME."""
+import attr
+
+from cached_property import cached_property
+from lxml.html import document_fromstring
+from navmazing import NavigateToAttribute
+from widgetastic_manageiq import (
+    BaseEntitiesView,
+    NonJSBaseEntity,
+    JSBaseEntity,
+    BaseListEntity,
+    BaseQuadIconEntity,
+    BaseTileIconEntity,
+    BreadCrumb,
+    ItemsToolBarViewSelector,
+    SummaryTable,
+    ManageIQTree
+)
+from widgetastic_patternfly import Dropdown, Accordion
+from widgetastic.widget import ParametrizedView, Text, View
+
+from cfme.base.login import BaseLoggedInPage
+from cfme.common import Taggable
+from cfme.exceptions import (
+    ItemNotFound,
+    StatsDoNotMatch,
+    RackStatsDoesNotContain,
+    ProviderHasNoProperty
+)
+from cfme.modeling.base import BaseEntity, BaseCollection
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigate_to, navigator
+from cfme.utils.log import logger
+from cfme.utils.pretty import Pretty
+from cfme.utils.providers import get_crud_by_name
+from cfme.utils.update import Updateable
+from cfme.utils.wait import wait_for
+
+
+@attr.s
+class PhysicalRack(BaseEntity, Updateable, Pretty, Taggable):
+    """Model of an Physical Rack in cfme.
+
+    Args:
+        name: Name of the physical rack.
+        custom_ident: The custom identifiter.
+        provider: The physical provider.
+
+    Usage:
+        myrack = PhysicalRack()
+        myrack.create()
+
+    """
+    pretty_attrs = ['name', 'custom_ident']
+
+    name = attr.ib()
+    provider = attr.ib(default=None)
+    custom_ident = attr.ib(default=None)
+    db_id = None
+
+    INVENTORY_TO_MATCH = ['power_state']
+    STATS_TO_MATCH = ['cores_capacity', 'memory_capacity']
+
+    def load_details_page(self):
+        """Load the PhysicalRack details page."""
+        view = navigate_to(self, "Details", use_resetter=False, wait_for_view=True)
```

Comment:
There is no resetter on either navigation class below, use_resetter=False is not doing anything here as a result, just FYI